src/components/organisms/AssetActions/Edit/index.tsx

```diff
@@ -60,13 +61,29 @@ export default function Edit({
   const { debug } = useUserPreferences()
   const { accountId } = useWeb3()
   const { ocean } = useOcean()
-  const { metadata, ddo, refreshDdo } = useAsset()
+  const { metadata, ddo, refreshDdo, price } = useAsset()
   const [success, setSuccess] = useState<string>()
   const [error, setError] = useState<string>()
   const [timeoutStringValue, setTimeoutStringValue] = useState<string>()
 
   const hasFeedback = error || success
 
+  async function updateFixedPrice(newPrice: number) {
+    const exchanges = await ocean.fixedRateExchange.searchforDT(
```

Comment:
do we need to do this expensive search? Cause it looks like we get the exchange ID from `price.address` for FRE assets https://github.com/oceanprotocol/market/blob/main/src/providers/Asset.tsx#L117

Reply:
yeah, good point, we don't